### PR TITLE
Aufgabe3

### PR DESCRIPTION
**Pull request by MariamaB**

### CODE REVIEW

File: 1/src/components/ListItem.vue [hunk 1]
```diff
@@ -41,24 +41,28 @@
...
       <div v-if="checkLoggedInUser">
-        <a v-if="this.todo.status" ><img title="Done" class="icon" src="../assets/done.png" /></a>
-        <a v-else @click="setTodoDone"><img title="Done_Pending" class="icon" src="../assets/done_pending.png" /></a>
+        <a v-if="this.todo.status">
+          <img title="Done" class="icon" src="../assets/done.png" />
+        </a>
+        <a v-else @click="setTodoDone">
+          <img title="Done_Pending" class="icon" src="../assets/done_pending.png" />
+        </a>
       </div>
```
Comment (roschaefer) [inline 1/src/components/ListItem.vue]: There is only indentation changes here, right? For your code-reviewer it's sometimes hard to see a difference in the implementation.

File: 1/src/components/ListItem.vue [hunk 2]
```diff
@@ -68,17 +72,16 @@ export default {
     return {
       onEdit: false,
       newTodo: {
-        message: '',
+        message: "",
```
Comment (roschaefer) [inline 1/src/components/ListItem.vue]: I'd prefer single quotes.

File: 1/src/components/ListItem.vue [hunk 3]
```diff
@@ -91,47 +94,51 @@ export default {
...
-        let user = this.users.find(u => (u.id === this.todo.assignedTo));
-        return user.loggedIn ? true : false;    
-      }else{
-        return false;
-      }   
-  }
+    checkLoggedInUser() {
+      if (this.todo.assignedTo === 0) return false;
+      let user = this.users.find(u => u.id === this.todo.assignedTo);
+      return user && user.loggedIn;
```
Comment (roschaefer) [inline 1/src/components/ListItem.vue]: So much easier to read.

File: 1/src/components/auth/Login.vue [hunk 4]
```diff
@@ -2,87 +2,102 @@
...
+  props: {},
   methods: {
-      async login(){
-         await this.$apollo.mutate({
-      mutation: gql`mutation ($email: String!, $password: String!) {
-        login(email: $email, password: $password) {
-          id
+    async login() {
+      console.log("Login mutation started");
+      const data = await this.$apollo.mutate({
```
Comment (roschaefer) [inline 1/src/components/auth/Login.vue]:
```suggestion
      const { dta: { login } } = await this.$apollo.mutate({
```

File: 1/src/components/auth/Login.vue [hunk 5]
```diff
@@ -2,87 +2,102 @@
...
-      //               this.id = '';
-      //               }
-                    
-      //           }).catch((error) => {
-      //               return error;
-      //           })
-      // }
+      });
+      console.log("Before if");
+      if (data.data.login != null || data.data.login != undefined) {
```
Comment (roschaefer) [inline 1/src/components/auth/Login.vue]:
```suggestion
      if (!login) {
```

See above :point_up: how I assigned the local variable `login`.

File: 1/src/components/auth/Login.vue [hunk 6]
```diff
@@ -2,87 +2,102 @@
...
-        email,
-        # password,
-        loggedIn
+    users: gql`
+      query {
+        users {
+          id
+          email
+          # password,
+          loggedIn
```
Comment (roschaefer) [inline 1/src/components/auth/Login.vue]: As I said: I find it quite suboptimal to return an attribute `loggedIn` on every user which indicates if that's the user which is logged in. Better have one mutation that's returning a `jwt` and the logged in `user`.

File: 1/src/_backend/jwt/.env [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+GRAPHQL_PORT=4000
+GRAPHQL_URI=http://localhost:4000
+CLIENT_URI=http://localhost:8080
+SMTP_HOST=
+SMTP_PORT=
+SMTP_IGNORE_TLS=true
+SMTP_USERNAME=
+SMTP_PASSWORD=
+
+JWT_SECRET="b/&&7b78BF&fv/Vd"
```
[the remark this answers is not in the document]
Comment (MariamaB) [reply 1/src/_backend/jwt/.env]: Unfortunatley we couldn't reproduce the diff correctly here are the links you requested for.

Link to test :  https://github.com/MariamaB/Systems-Development-and-Frameworks/blob/develop/1/tests/unit/todo.spec.js

Link to backend files : https://github.com/MariamaB/Systems-Development-and-Frameworks/tree/develop/1/src/_backend

[the remark this answers is not in the document]
Comment (roschaefer) [reply 1/src/_backend/jwt/.env]: I cannot give a review on the code. But you're missing `await` here: 

https://github.com/MariamaB/Systems-Development-and-Frameworks/blob/develop/1/tests/unit/todo.spec.js#L60

That's pretty dangerous. Please do mutant testing when you write or refactor those tests.

Apart from that, the tests look OK to me :+1: 

By the way, here's how you could construct the desired diff:
1. Identify a commit id after you submitted exercise 2
2. Identify a commit after you submitted exercise 3
3. Create a `target_branch` from 1)
4. Create a `source_branch` from 2)
5. Create a PR from `target_branch`<-`source_branch`

**Pull request merged**